src/observable/fromEvent.ts

```diff
@@ -4,37 +4,63 @@ import {errorObject} from '../util/errorObject';
 import {Subscription} from '../Subscription';
 import {Subscriber} from '../Subscriber';
 
+export type EventEmmitterLike1 = {
+  addListener: (eventName: string, handler: Function) => void;
+  removeListener: (eventName: string, handler: Function) => void;
+};
+function isEventEmmitterLike1(sourceObj: any): sourceObj is EventEmmitterLike1 {
+  return typeof sourceObj.addListener === 'function' && typeof sourceObj.removeListener === 'function';
+}
+
+export type EventEmmitterLike2 = {
+  on: (eventName: string, handler: Function) => void;
+  off: (eventName: string, handler: Function) => void;
+};
+function isEventEmmitterLike2(sourceObj: any): sourceObj is EventEmmitterLike2 {
+  return typeof sourceObj.on === 'function' && typeof sourceObj.off === 'function';
+}
+
+function isEventTargetLike(sourceObj: any): sourceObj is EventTarget {
+  return typeof sourceObj.addEventListener === 'function' && typeof sourceObj.removeEventListener === 'function';
+}
+
+function isNodeListOrHTMLCollection(sourceObj: any): sourceObj is NodeList | HTMLCollection {
+  let tag = sourceObj.toString();
+  return tag === '[object NodeList]' || tag === '[object HTMLCollection]';
+}
+
+export type EventTargetLike = EventTarget | EventEmmitterLike1 | EventEmmitterLike2 | NodeList | HTMLCollection;
+
+function setupSubscription<T>(sourceObj: EventTargetLike, eventName: string, handler: Function, subscriber: Subscriber<T>) {
```

Comment:
I also don't have historical detail why it's originally designed in those way, so curious about way you chose. Personally though function does not depends on instance of class, it still makes sense to have it organized under class. @blesh , would you mind explain detail?